samtranslator/intrinsics/actions.py

```diff
@@ -561,9 +561,9 @@ def resolve_parameter_refs(self, input_dict, parameters):
             return input_dict
 
         if (
-            map_name not in parameters
-            or top_level_key not in parameters[map_name]
-            or second_level_key not in parameters[map_name][top_level_key]
+            not parameters.get(map_name)
+            or not parameters.get(map_name, {}).get(top_level_key)
+            or not parameters.get(map_name, {}).get(top_level_key, {}).get(second_level_key)
```

Comment:
I think it is better to validate that these values are instance of Dict, as for invalid template, these values can exist as strings or None values instead of dictionary